src/core/config.c

```diff
@@ -198,38 +198,34 @@ bool mCoreConfigSaveVFile(const struct mCoreConfig* config, struct VFile* vf) {
 	return ConfigurationWriteVFile(&config->configTable, vf);
 }
 
-void mCoreConfigMakePortable(const struct mCoreConfig* config) {
+void mCoreConfigMakePortable(const struct mCoreConfig* config, const char* path) {
 	struct VFile* portable = NULL;
 	char out[PATH_MAX];
-	mCoreConfigPortablePath(out, sizeof(out));
+	mCoreConfigPortableIniPath(out, sizeof(out));
 	if (!out[0]) {
 		// Cannot be made portable
 		return;
 	}
 	portable = VFileOpen(out, O_WRONLY | O_CREAT);
 	if (portable) {
+		if (path && path[0]) {
+			struct Configuration portableConfig;
+			ConfigurationInit(&portableConfig);
+			ConfigurationSetValue(&portableConfig, "Portable", "Path", path);
+			ConfigurationWriteVFile(&portableConfig, portable);
+		}
```

Comment:
You should have an else clause that resets the value to an empty string, otherwise it won't overwrite a previous setting, if applicable.